python/tvm/target/target.py

```diff
@@ -413,79 +413,117 @@ def bifrost(model="unknown", options=None):
     return Target(" ".join(["opencl"] + opts))
 
 
-def hexagon(cpu_ver="v66", sim_args=None, llvm_args=None, hvx=128):
+def hexagon(cpu_ver="v66", **kwargs):
```

Comment:
Thanks @zxybazh for the detailed feedback.  You are making valid points, but I think I should share more context of why the code looks like this.

In our downstream repository we have a few more options for the Hexagon target that, for one reason or another, we cannot yet make public.  We have reorganized the `tvm.target,hexagon()` so that we don't need to change the interface every time we add a new flag---this is the reason to use the keyword arguments instead of explicit named arguments.
The code downstream has diverged from the upstream version, and we would like to keep these two close to reduce the risk of conflicts (when integrating upstream code to our downstream repo).  Essentially, what this patch is, is the downstream code with the non-public parts deleted.  This is why several places look like they could be simplified, but if we did simplify them upstream, it would create divergence again.
I'd like to ask that this code is accepted without removing `args` since the downstream code has more fields in it, and is organized around handling the contents of the target configuration represented by `args`.

Reply:
I'll revisit the other comments a bit later.  The general idea there was to create sim/llvm/etc strings from the "target configuration", which is represented by `args` and `cpu_ver`.  This is why all the string-generating functions take the same arguments, even if some of these arguments aren't necessary, or are already available without parameters.

Reply:
Thanks for the clarification. It would be fine to me if you want to keep the `args` variable this way. Would appreciate a few lines of docs explaining the reason. Please feel free to the remove the argument related issues I proposed.

Reply:
I renamed `args` to `config`, hopefully it will better reflect the purpose of the variable.